xio-core/src/test/java/com/xjeffrose/xio/config/ConfigReloaderUnitTest.java

```diff
@@ -39,22 +39,53 @@ String createConfig(String value) throws IOException, FileNotFoundException {
     }
   }
 
+  public static class TrivialState {
+    TrivialConfig config;
+
+    TrivialState(TrivialConfig config) {
+      this.config = config;
+    }
+
+    public void update(TrivialConfig oldValue, TrivialConfig newValue) {
+      this.config = newValue;
+    }
+
+    public void fireUpdated() {}
+  }
+
+  @Test
+  public void testWiring() throws Exception {
+    ScheduledExecutorService executor = new ScheduledThreadPoolExecutor(1);
+    ConfigReloader<TrivialConfig> reloader = new ConfigReloader<>(executor, TrivialConfig::new);
+    TrivialConfig config = reloader.init(createConfig("10"));
+    TrivialState state =
+        new TrivialState(config) {
+          @Override
+          public void fireUpdated() {
+            assertEquals(20, this.config.limit);
+            executor.shutdown();
+          }
+        };
+    assertEquals(10, config.limit);
+    reloader.start(state::update);
+
+    Thread.sleep(2000);
```

Comment:
CountDownLatch might be better than a sleep

Reply:
I just tried to replace the sleeps with countdown latches, and that's when I was brutally reminded that we're checking the modification date of a file in the implementation. To do this right without sleeps we'll need to mock out that modification check. Not super concerned about this at the moment we can throw this into a tech debt backlog.